cmd/s3-gw/app.go

```diff
@@ -184,36 +187,40 @@ func (a *App) initAPI(ctx context.Context) {
 }
 
 func (a *App) initMetrics() {
-	gateMetricsProvider := newGateMetrics(neofs.NewPoolStatistic(a.pool))
+	gateMetricsProvider := newGateMetrics(neofs.NewPoolStatistic(a.poolStat))
 	gateMetricsProvider.SetGWVersion(version.Version)
 	a.metrics = newAppMetrics(a.log, gateMetricsProvider, a.cfg.GetBool(cfgPrometheusEnabled))
 }
 
-func (a *App) initResolver() {
-	var err error
-	a.bucketResolver, err = resolver.NewBucketResolver(a.getResolverConfig())
+func (a *App) initResolver(ctx context.Context) {
+	cl, err := rpcClient(ctx, a.cfg.GetString(cfgRPCEndpoint))
 	if err != nil {
-		a.log.Fatal("failed to create resolver", zap.Error(err))
+		a.log.Fatal("rpcclient", zap.Error(err))
 	}
-}
 
-func (a *App) getResolverConfig() ([]string, *resolver.Config) {
-	resolveCfg := &resolver.Config{
-		NeoFS:      neofs.NewResolverNeoFS(a.pool),
-		RPCAddress: a.cfg.GetString(cfgRPCEndpoint),
+	nnsHash, err := sidechain.SystemContractHash(cl, sidechain.NNSContract)
```

Comment:
Do we have a similar functionality inside contracts? I mean resolving system contract hashes

Reply:
It's a part of a regular RPC client, so this shouldn't be in contracts.

Reply:
Ok, I don't mind having such a function